.github/workflows/pr.yml

```diff
@@ -39,17 +39,19 @@ jobs:
       - run: yarn test
 
       - name: Archive test results
-        uses: actions/upload-artifact@v2 # upload test results
+        uses: actions/upload-artifact@v4 # upload test results
         if: always() # run this step even if previous step failed
         with:
           name: test-results
           path: testResults/junit.xml
+          overwrite: true
```

Comment:
If you want to keep artifacts from each run from the matrix, you would need to make the artifact name unique based off the run. This is due to breaking changes in V4 of the `upload-artifact` task - https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes

You can take a look at an example of unique naming in this PR - https://github.com/microsoftgraph/msgraph-metadata/pull/537

Reply:
Thanks @andrueastman, trying that.